torchtitan/config_manager.py

```diff
@@ -100,6 +100,18 @@ def __init__(self):
             default=10,
             help="How often to collect profiler traces, in iterations",
         )
+        self.parser.add_argument(
+            "--profiling.enable_memory_snapshot",
+            action="store_true",
+            default=False,
+            help="Whether to dump memory snapshot",
+        )
+        self.parser.add_argument(
+            "--profiling.memory_snapshot_folder",
```

Comment:
nit: please rename it `save_memory_snapshot_folder` to be consistent with `save_traces_folder` and `save_tb_folder`.